src/Stryker.Core/Stryker.Core/Initialisation/InputFileResolver.cs

```diff
@@ -62,25 +62,28 @@ public ProjectInfo ResolveInput(string currentDirectory, string projectName)
         /// </summary>
         /// <param name="path"></param>
         /// <returns></returns>
-        private FolderComposite FindInputFiles(string path)
+        private FolderComposite FindInputFiles(string path, List<string> filesToExclude)
         {
-            var folderComposite = new FolderComposite()
+            var folderComposite = new FolderComposite
             {
                 Name = Path.GetFullPath(path)
             };
-            foreach (var folder in _fileSystem.Directory.EnumerateDirectories(path).Where(x => !_foldersToIgnore.Contains(Path.GetFileName(x))))
+
+            foreach (var folder in _fileSystem.Directory.EnumerateDirectories(path).Where(x => !_foldersToExclude.Contains(Path.GetFileName(x))))
             {
-                folderComposite.Add(FindInputFiles(folder));
+                folderComposite.Add(FindInputFiles(folder, filesToExclude));
             }
-            foreach (var file in _fileSystem.Directory.GetFiles(path, "*.cs", SearchOption.TopDirectoryOnly))
+
+            foreach (var file in _fileSystem.Directory.GetFiles(path, "*.cs", SearchOption.TopDirectoryOnly).Except(filesToExclude))
```

Comment:
I have not tested this yet but I suddenly doubt this will work. Later in the process we feed the Roslyn compiler the abstract syntax trees of all files. Roslyn needs this to compile correctly. I guess that if an important file is excluded Stryker cannot compile the program because of missing types.

I think it's better to read the file and add it to the composite. But we mark it as excluded, to skip it later from exclusion. This also has an advantage. We could display in the final console report that this file was excluded. Using this current method we can't do that.

Reply:
Hmmm, it's still getting heavier this pull request :-D ... haha but I understand what you mean. Quite funny though, due to the fact that you have in an earlier comment suggested to do it like this :-). Anyway I'll be going back to the drawing board!

Reply:
Yeah I know I'm sorry about that. I was just looking at your code style the first time. Now I took some time to look at the functionallities. I like how you are so motivated to get this right! The feature would be more awesome this way too if you ask me :D

Reply:
No sorry man! Just getting the job done :-). I've convinced many people to use Stryker.net so yea ... it should be good :-)

Reply:
@richardwerkman Aight, got it :-), you can check it out!